### PR TITLE
add support to include `ResponsiveImages` based on condition

### PR DESCRIPTION
**Pull request by ctf0**
allow `withResponsiveImages` to be added based on condition, this is useful if you have multiple media collections with the same characteristics but you need the responsive images for only one ex.

```php
public function registerMediaCollections(): void
{
    $singles = ['cover', 'scan'];

    foreach ($singles as $item) {
        $this->addMediaCollection($item)
            ->useFallbackUrl(asset('images/ph.png'))
            ->useFallbackPath(public_path('images/ph.png'))
            ->withResponsiveImages($item != 'scan')
            ->singleFile();
    }
}
```

### CODE REVIEW

**Comment by freekmurze** [comment]
I like the requested functionality, but not the implementation. Instead of changing the existing function, let's add `withResponsiveImagesIf` that accepts a boolean.

Also add a test to make sure that it is working.

**Comment by freekmurze** [comment]
Feel free to work further on this PR.

**Comment by ctf0** [comment]
i thought you dont want to bloat the lib with extra methods, specially that the new change is backward compatible with the current version usage.

i was also thinking of turning the variable into a closure to give more flexibility, what u think ?

**Comment by freekmurze** [comment]
Let's support both a bool and a callable

**Comment by ctf0** [comment]
np

**Comment by ctf0** [comment]
as for tests, am not sure where to add the test for this one, i found `withResponsiveImages` in 8 test file.

File: src/MediaCollections/MediaCollection.php [hunk 1]
```diff
@@ -120,4 +120,15 @@ public function withResponsiveImages(): self
 
         return $this;
     }
+
+    public function withResponsiveImagesIf($condition): self
+    {
+        if (is_callable($condition)) {
```
Comment (freekmurze) [inline src/MediaCollections/MediaCollection.php:126]: Use a ternary here, so we can lose the `else` statement.

**Review by freekmurze — CHANGES_REQUESTED** [review]

File: src/MediaCollections/MediaCollection.php (hunk 1 above)
Comment (ctf0) [reply src/MediaCollections/MediaCollection.php:126]: will do, was also thinking of adding an exception to make sure we only get this 2 types.

**Comment by freekmurze** [comment]
> as for tests, am not sure where to add the test for this one, i found `withResponsiveImages` in 8 test file.

Put them where you think is a logical place

**Comment by ctf0** [comment]
any updates ?

**Comment by freekmurze** [comment]
Thanks!

**Comment by rcerljenko** [comment]
Can we also have this?
https://github.com/spatie/laravel-medialibrary/discussions/2083

**Pull request merged**